sealedenumprocessor/src/test/kotlin/com/livefront/sealedenum/GenericSealedClassTests.kt

```diff
@@ -9,18 +9,24 @@ sealed class OneTypeParameterSealedClass<out T> {
     object SecondObject : OneTypeParameterSealedClass<String>()
 
     object ThirdObject : OneTypeParameterSealedClass<Nothing>()
+
+    companion object
```

Comment:
There's one last topic I want to raise, and it's the auto-generation for all sealed classes.  

I think it makes total sense as a library capability to generate `sealed-enum` code without needing the `@GenSealedEnum` as a prompt.  My bias would be to have that `opt-in` only.  If they're _really_ sure they want to run the code generation checks over every sealed class, they can.

A few benefits to opt-in:

- devs love lightweight 
This is my bias showing again, but as I've been working with the `target` codebase and its sprawling build pipeline, anything that adds any time to the build is _loathed_, even for tools we love like moshi and dagger.  The fastest code continues to be no code at all.  And though it surely will improve in the future, `kapt` and AS have cache poisoning problems that come up when switching branches even when code generation is working perfectly.

- discoverability
we can let the consuming developer match their domain problem with the `sealed-enum`, and anyone reading a domain model in the consuming project can click through and see what's happening, and what this "sealed enum" thing is all about.

Reply:
Totally agree, it's already opt-in 😄 .

The reason why it's happening in the `sealedenumprocessor` tests is the `kapt` section:
```kotlin
kapt {
    includeCompileClasspath = false
    arguments {
        arg("sealedenum.autoGenerateSealedEnums", "true")
    }
}
```

This is also the reason why there's the assemble warning, since that flag is being applied during normal `kapt` and nothing is picking it up. I couldn't find a way to have "test only" arguments so-to-say.

(I also just added the `includeCompileClasspath` to remove some warnings when performing tests).

I also could honestly be convinced to remove auto generation entirely as a feature, in favor of having to have the explicit annotations.

Reply:
Ah, gotcha! Thanks for clarifying.

> I also could honestly be convinced to remove auto generation entirely as a feature, in favor of having to have the explicit annotations.

I would be on board for that.

Reply:
Agreed! The autogeneration feature was removed in c391838